### PR TITLE
GCC 10 flags, compiler check, and settable flags

### PR DESCRIPTION
**Pull request by kgerheiser**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -2,15 +2,19 @@
 
 if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
   set(CMAKE_Fortran_FLAGS
-      "-g -xHOST -traceback -free -convert big_endian -assume byterecl")
+      "-g -xHOST -traceback -free -convert big_endian -assume byterecl ${CMAKE_Fortran_FLAGS}")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: is `-xHOST` required here?

Comment (climbfuji) [reply src/CMakeLists.txt]: Please remove or replace with
```
-axSSE4.2,AVX,CORE-AVX2,CORE-AVX512
```
or
```
-axSSE4.2,AVX,CORE-AVX2
```

@DusanJovic-NOAA any preference on adding AVX512 or not?

Comment (kgerheiser) [reply src/CMakeLists.txt]: I notice that it's all the I/O libraries that have `-xHost` (sigio, sfcio, gfsio). Is there a reason for that I'm missing? I would think vectorization and other optimizations would have little effect on a library like this.

Comment (DusanJovic-NOAA) [reply src/CMakeLists.txt]: I think we should not hard code any architecture specific flags.

**Comment by DusanJovic-NOAA** [comment]
I see gcc-10 flags are being added to almost all libraries. I think we should add them only to libraries that need them.

**Comment by kgerheiser** [comment]
I agree, that actually just came into my mind as I'm on the last one.

I'll check out where the actual failures are and fix it.

**Comment by aerorahul** [comment]
> I see gcc-10 flags are being added to almost all libraries. I think we should add them only to libraries that need them.

**Comment by kgerheiser** [comment]
Not required for this library. I removed the GCC 10 check.

**Review by aerorahul — APPROVED** [review]

**Comment by DusanJovic-NOAA** [comment]
And some (bacio for example) need just '`-fallow-argument-mismatch`' not '`-fallow-invalid-boz`'.

**Comment by aerorahul** [comment]
@kgerheiser I did not check bufr library which flag is required for gfortran-10 because I don't have a gcc-10 installation.
Would you mind checking that too? Thanks!

**Comment by DusanJovic-NOAA** [comment]
Do we want '-mkl' in nemsio (intel)?

**Comment by DusanJovic-NOAA** [comment]
> Do we want '-mkl' in nemsio (intel)?

I see comment in nemsio PR. Sorry.

**Pull request merged**